infra/examples-dev/aws-all/variables.tf

```diff
@@ -294,3 +294,14 @@ variable "todos_as_local_files" {
   description = "whether to render TODOs as flat files"
   default     = true
 }
+
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  validate_salesforce_domain         = (var.salesforce_domain == null || var.salesforce_domain == "") && contains(var.enabled_connectors, "salesforce")
+  validate_salesforce_domain_message = "The salesforce_domain var should be populated if enabled."
+  validate_salesforce_domain_check = regex(
```

Comment:
Weird validation, open if there are other way. The problem is that `validation` can only work with same variable; in this case it involves two vars (domain + enabled connectors). 

It could be solved in the future if we have an object for all stuff related with connectors, so we don't need to add variables in all the places everytime we want to add/modify something and maybe it would be cleaner.

Reply:
yeah due to this limitation in Terraform I think best approach is to validate that `var.salesforce_domain`, if provided, is domain without path or protocol.

Then validate `TARGET_HOST` in java as you have. 

additionally, we could also use https://developer.hashicorp.com/terraform/language/checks on `TARGET_HOST`; but as only from 1.5+ may need to wait on that, as many people still on 1.3x or 1.4x